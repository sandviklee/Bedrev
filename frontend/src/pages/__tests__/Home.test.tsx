import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import HomePage from "../Home.tsx";

describe("Home page", () => {
  test("renders correctly", () => {
    render(
      <Router>
        <HomePage />
      </Router>,
    );
    expect(screen.getByText("Se gjennom bedrifter,")).toBeInTheDocument();
    expect(
      screen.getByText("finn din neste arbeidsgiver!"),
    ).toBeInTheDocument();
    expect(screen.getByText("Søk gjennom bedrifter")).toBeInTheDocument();
    expect(
      screen.getByText("Skriv tilbakemeldinger til Bedrifter!"),
    ).toBeInTheDocument();
    expect(screen.getByText("Anonym om Bekk Consulting")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <Router>
        <HomePage />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Has navbar", () => {
    render(
      <Router>
        <HomePage />
      </Router>,
    );
    expect(screen.getByText("Logg Inn")).toBeInTheDocument();
  });

  test("Has footer", () => {
    render(
      <Router>
        <HomePage />
      </Router>,
    );
    expect(screen.getByText("©2023")).toBeInTheDocument();
  });

  test("Button navigates to correct route", () => {
    render(
      <Router>
        <HomePage />
      </Router>,
    );
    const button = screen.getByText("Søk gjennom bedrifter").closest("a");
    expect(button?.getAttribute("href")).toBe("/sok-bedrifter");
  });
});
