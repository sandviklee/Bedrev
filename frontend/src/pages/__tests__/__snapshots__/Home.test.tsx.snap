// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex flex-row w-full h-[120px] items-center px-[8%] text-base font-whyte"
  >
    <a
      href="/"
    >
      <img
        alt="companylogo"
        class="w-[150px]"
        src="/project2/bedrevLogo.svg"
      />
    </a>
    <div
      class="sm:visible sm:flex flex-row h-full w-full items-center justify-end hidden"
    >
      <a
        class="flex items-center justify-center h-[100%] w-[120px] font-whyte"
        href="/"
      >
        <div
          class="pb-1.5 pr-2"
          style="visibility: visible;"
        >
          <div
            class="w-2 h-2 rounded-xl bg-purple"
          />
        </div>
        <p>
          Hjem
        </p>
      </a>
      <a
        class="flex items-center justify-center h-[100%] w-[120px] font-whyte"
        href="/sok-bedrifter"
      >
        <div
          class="pb-1.5 pr-2"
          style="visibility: hidden;"
        >
          <div
            class="w-2 h-2 rounded-xl bg-purple"
          />
        </div>
        <p>
          Søk bedrifter
        </p>
      </a>
      <a
        class="flex items-center justify-center h-[100%] w-[120px] font-whyte"
        href="/logg-inn"
      >
        <div
          class="pb-1.5 pr-2"
          style="visibility: hidden;"
        >
          <div
            class="w-2 h-2 rounded-xl bg-purple"
          />
        </div>
        <p>
          Logg Inn
        </p>
      </a>
    </div>
    <div
      class="sm:hidden flex w-full h-full items-center justify-end visible pb-4"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        data-state="closed"
        id="radix-:r2:"
        type="button"
      >
        <div
          aria-expanded="false"
          class="hamburger-react"
          role="button"
          style="cursor: pointer; height: 48px; position: relative; transition: 0.4s cubic-bezier(0, 0, 0, 1); user-select: none; width: 48px; outline: none; transform: none;"
          tabindex="0"
        >
          <div
            style="background: currentColor; height: 3px; left: 8px; position: absolute; width: 32px; top: 13px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
          />
          <div
            style="background: currentColor; height: 3px; left: 8px; position: absolute; width: 32px; top: 23px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
          />
          <div
            style="background: currentColor; height: 3px; left: 8px; position: absolute; width: 32px; top: 33px; transition: 0.4s cubic-bezier(0, 0, 0, 1); transform: none;"
          />
        </div>
      </button>
    </div>
  </nav>
  <div
    class=" h-auto w-full overflow-hidden font-whyte"
  >
    <section
      class="sm:flex sm:flex-row items-center justify-center gap-[80px] px-[8%] sm:pt-[4%] pb-[8%]"
    >
      <div
        class="leading-[40px] sm:text-3xl text-2xl"
      >
        <p>
          Se gjennom bedrifter,
        </p>
        <p>
          finn din neste arbeidsgiver!
        </p>
        <p
          class="max-w-[480px] pt-4 text-base leading-[35px] py-6"
        >
          Er du på jakt etter din neste jobbmulighet, eller bare nysgjerrig på hvordan det er å jobbe hos en bestemt bedrift? Hos 
          <span
            class=" font-bold"
          >
            bedrev
          </span>
           får du tilgang til tilbakemeldinger fra ansatte, innsyn i lønnsvilkår, bilder av kontorområder, og mye mer. Søk, utforsk og finn din drømmebedrift med oss i dag!
        </p>
        <a
          href="/sok-bedrifter"
        >
          <button
            class="bg-white text-black hover:bg-white hover:text-black rounded-sm border-black border-2 transition-all py-2 w-fit text-sm px-[8%] group transition hover:bg-primary border-solid"
          >
            <div
              class="flex flex-row gap-2 justify-center pt-1 h-full px-6 align-middle"
            >
              Søk gjennom bedrifter
              <svg
                class="text-lg group-hover:translate-x-2 transition-transform delay-75 pb-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
                />
              </svg>
            </div>
          </button>
        </a>
      </div>
      <div
        class="items-center justify-end"
      >
        <img
          alt="Two people working"
          class=" sm:w-min-[520px] w-min-[300px]"
          src="/project2/homepageArt.svg"
        />
      </div>
    </section>
    <section
      class="h-[500px] bg-purple-lighter"
    >
      <div
        class="w-full justify-center text-center py-[30px] sm:leading-[30px]"
      >
        <p
          class="sm:text-2xl text-xl font-medium text-purple"
        >
          Skriv tilbakemeldinger til Bedrifter!
        </p>
        <p
          class="sm:text-base text-sm sm:leading-[30px] tracking-wider"
        >
          Legg inn en tilbakemelding da vell!
          <br />
          Vi har 
          <span
            class="underline-offset-1 underline"
          >
            Norges toppbedrifter!
          </span>
        </p>
      </div>
      <div
        class="flex justify-center gap-4 px-2"
      >
        <div
          class="px-[20px] py-[20px] flex-row h-[250px] w-[350px] bg-purple-light rounded-md font-whyte"
        >
          <p
            class="sm:leading-[40px] leading-[25px] text-lg font-medium"
          >
            Utrolig sted å jobbe
          </p>
          <p
            class="leading-[30px] text-sm  h-[140px]"
          >
            Elsker å jobbe hos Kantega. Lokalene er store, luftige og store vinduer gjør at man alltid har dagslys. I tillegg blir det servert varm mat i kantina hver dag.
          </p>
          <p
            class="py-2 h-full w-full text-end text-base font-medium"
          >
            Anonym om Kantega
          </p>
        </div>
        <div
          class="px-[20px] py-[20px] flex-row h-[250px] w-[350px] bg-purple-light rounded-md font-whyte"
        >
          <p
            class="sm:leading-[40px] leading-[25px] text-lg font-medium"
          >
            Fantastisk
          </p>
          <p
            class="leading-[30px] text-sm  h-[140px]"
          >
            Liker kantinen veldig godt! Godt arbeidsmiljø, nye utfordringer hver dag!
          </p>
          <p
            class="py-2 h-full w-full text-end text-base font-medium"
          >
            Anonym om Bekk Consulting
          </p>
        </div>
      </div>
    </section>
  </div>
  <footer
    class="flex flex-row w-full sm:h-[180px] h-[120px] bg-purple items-center px-[8%] font-whyte"
  >
    <a
      href="/"
    >
      <div
        class="flex flex-row items-center justify-center gap-2"
      >
        <img
          alt="companylogo"
          class="pt-2 sm:w-[150px] w-[100px]"
          src="/project2/bedrevLogoWhite.svg"
        />
        <p
          class="sm:text-lg text-base text-white-paper pt-2"
        >
          ©2023
        </p>
      </div>
    </a>
    <div
      class="flex flex-row h-full w-full pt-[6px] items-center justify-end sm:gap-1 text-white-paper sm:text-base text-sm"
    >
      <a
        class="flex items-center justify-center align-center h-[100%] sm:w-[105px] w-[90px]"
        href="/"
      >
        <p>
          Hjem
        </p>
      </a>
      <a
        class="flex items-center justify-center align-center h-[100%] sm:w-[105px] w-[90px]"
        href="/sok-bedrifter"
      >
        <p>
          Søk bedrifter
        </p>
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
